ldap/servers/slapd/regex.c

```diff
@@ -123,33 +125,35 @@ slapi_re_exec(Slapi_Regex *re_handle, const char *subject, time_t time_up)
 int32_t
 slapi_re_exec_nt(Slapi_Regex *re_handle, const char *subject)
 {
-    int32_t rc;
+    int32_t rc = 0;
 
     if (NULL == re_handle || NULL == re_handle->re_pcre || NULL == subject) {
         return LDAP_PARAM_ERROR;
     }
 
-    if (NULL == re_handle->re_ovector) {
-        re_handle->re_oveccount = OVECCOUNT;
-        re_handle->re_ovector = (int *)slapi_ch_malloc(sizeof(int) * OVECCOUNT);
+    if (re_handle->match_data) {
+        pcre2_match_data_free(re_handle->match_data);
+    }
+    re_handle->match_data = pcre2_match_data_create_from_pattern(re_handle->re_pcre, NULL);
+
+    if (re_handle->mcontext == NULL) {
+        re_handle->mcontext = pcre2_match_context_create(NULL);
+        pcre2_set_match_limit(re_handle->mcontext, OVEC_MATCH_LIMIT);
     }
 
-    rc = pcre_exec(re_handle->re_pcre,       /* the compiled pattern */
-                   NULL,                     /* no extra data */
-                   subject,                  /* the subject string */
-                   strlen(subject),          /* the length of the subject */
-                   0,                        /* start at offset 0 in the subject */
-                   0,                        /* default options */
-                   re_handle->re_ovector,    /* output vector for substring info */
-                   re_handle->re_oveccount); /* number of elems in the ovector */
+    rc = pcre2_match(re_handle->re_pcre,
+                     (PCRE2_SPTR)subject,
+                     strlen(subject),
+                     0,
```

Comment:
The comments for what each argument was for was nice to have before, could be good to re-add them.

Reply:
> The comments for what each argument was for was nice to have before, could be good to re-add them.

Fixed